packages/m2m-gateway/test/api/agreements/createAgreement.test.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi } from "vitest";
+import { generateId } from "pagopa-interop-models";
+import { generateToken } from "pagopa-interop-commons-test";
+import { AuthRole, authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
+import { api, mockAgreementService } from "../../vitest.api.setup.js";
+import { appBasePath } from "../../../src/config/appBasePath.js";
+import { getMockedApiAgreement } from "../../mockUtils.js";
+import { toM2MGatewayApiAgreement } from "../../../src/api/agreementApiConverter.js";
+import {
+  missingMetadata,
+  resourcePollingTimeout,
+} from "../../../src/model/errors.js";
+
+describe("POST /agreements router test", () => {
+  const mockAgreementSeed: m2mGatewayApi.AgreementSeed = {
+    eserviceId: generateId(),
+    descriptorId: generateId(),
+  };
```

Comment:
Let's add a full seed
```suggestion
  const mockAgreementSeed: m2mGatewayApi.AgreementSeed = {
    eserviceId: generateId(),
    descriptorId: generateId(),
    delegationId: generateId()
  };
```